logic_tree_tools.py

```diff
@@ -0,0 +1,972 @@
+    }
+
+    modified_logic_tree_pairs = []
+
+    source_logic_tree = copy.deepcopy(initial_logic_tree_pair.source_logic_tree)
+    ground_motion_logic_tree = copy.deepcopy(
+        initial_logic_tree_pair.ground_motion_logic_tree
+    )
+
+    modified_logic_tree_pair = copy.deepcopy(initial_logic_tree_pair)
+
+    short_tectonic_region_types_for_ground_motion_logic_tree_note = [
+        trt_short_lookup_dict[trt] for trt in tectonic_region_type_set
+    ]
+    short_tectonic_region_types_for_source_logic_tree_note = copy.deepcopy(
+        short_tectonic_region_types_for_ground_motion_logic_tree_note
+    )
+
+    for short_trt_index, short_trt in enumerate(
+        short_tectonic_region_types_for_source_logic_tree_note
+    ):
+        if short_trt == "INTER":
+            short_tectonic_region_types_for_source_logic_tree_note[short_trt_index] = (
+                f"INTER_{which_interface}"
+            )
+
+    modified_source_logic_tree = select_branch_sets_given_tectonic_region_type(
+        source_logic_tree, tectonic_region_type_set, which_interface
+    )
+    modified_ground_motion_logic_tree = select_branch_sets_given_tectonic_region_type(
+        ground_motion_logic_tree, tectonic_region_type_set
+    )
+
+    modified_logic_tree_pair.source_logic_tree_note += f"tectonic_region_type_set:[{' '.join(short_tectonic_region_types_for_source_logic_tree_note)}] > "
+
+    modified_logic_tree_pair.ground_motion_logic_tree_note += f"tectonic_region_type_set:[{' '.join(short_tectonic_region_types_for_ground_motion_logic_tree_note)}] > "
+
+    modified_logic_tree_pair.source_logic_tree = copy.deepcopy(
+        modified_source_logic_tree
+    )
+    modified_logic_tree_pair.ground_motion_logic_tree = copy.deepcopy(
+        modified_ground_motion_logic_tree
+    )
+
+    modified_logic_tree_pairs.append(modified_logic_tree_pair)
+
+    return modified_logic_tree_pairs
+
+
+def print_info_about_logic_tree(logic_tree: Union[SourceLogicTree, GMCMLogicTree]):
+    """
+    Prints information about a logic tree.
+
+    This function prints details about the type of logic tree (SourceLogicTree or GMCMLogicTree),
+    the number of branch sets it contains, and information about each branch set.
+
+    Parameters
+    ----------
+    logic_tree : Union[SourceLogicTree, GMCMLogicTree]
+        The logic tree for which information is to be printed.
+    """
+
+    print("")  # Add a blank line for readability
+
+    if isinstance(logic_tree, SourceLogicTree):
+        print("Logic tree is a SourceLogicTree")
+
+    if isinstance(logic_tree, GMCMLogicTree):
+        print("Logic tree is a GMCMLogicTree")
+
+    print(f"Logic tree has {len(logic_tree.branch_sets)} branch sets")
+
+    for branch_set_index, branch_set in enumerate(logic_tree.branch_sets):
+
+        print(
+            f"Branch set index {branch_set_index} has name {branch_set.long_name} ({branch_set.short_name}) and contains {len(branch_set.branches)} branches"
+        )
```

Comment:
Maybe instead of the `print_info_about_logic_tree`, consider adding a `__repr__` method to the logic tree classes and then the user can just type

```
>>> logic_tree
# output of the info function
```

to get this information in a more Pythonic way?

Reply:
This is a good suggestion but the logic tree classes already have __repr__ methods in the GNS code that I would prefer to leave unchanged for simplicity